OONIProbeUnitTests/ResultTest.m

```diff
@@ -0,0 +1,58 @@
+#import <XCTest/XCTest.h>
+#import "Result.h"
+#define REPORT_ID @"REPORT_ID"
+
+@interface ResultTest : XCTestCase
+
+@end
+
+@implementation ResultTest
+Result *result;
+Measurement *measurement;
+
+- (void)setUp {
+    // Put setup code here. This method is called before the invocation of each test method in the class.
+    [[[Result query] fetch] remove];
+    result = [Result new];
+    measurement = [Measurement new];
+    measurement.result_id = result;
+    [measurement save];
+}
+
+- (void)tearDown {
+    // Put teardown code here. This method is called after the invocation of each test method in the class.
+}
+
+- (void)testNoMeasurements {
+    //No measurement, isEveryMeasurementUploaded true
```

Comment:
This is now wrong: there is a measurement in the database

Reply:
read it again, i use [Result new] not the previous initialized result

Reply:
So, to be really really really pick, I'd drop the global object and just write something like:

```ObjectiveC
- (void)testNoMeasurements {
  Result *result = [Result new];
  Measurement *measurement = [Measurement new];
  measurement.result_id = result;
  [measurement save];
  // Testing code...
}
```

Also, I still maintain that `testNoMeasurement` is not the correct name. You clearly have one saved measurement in the database, rather than no measurement. Please:

1. drop the globals

2. make it obvious what is the state of each test case (implicit is in general bad and here it seems very bad to the point that, unless I'm seriously mistaken, we have a "no measurement" test that contains measurements).

3. as an additional mechanism for robustness, please count that the number of all measurements is consistent with the number of measurements you actually inserted (so I don't have to guess but the code testifies that).

Thanks <3

Reply:
> You clearly have one saved measurement in the database, rather than no measurement.

This is fake news. The code is actually different than I though. But I guess this is also because I didn't expect the code to do what it is doing. Anyway, my apologies.